### PR TITLE
Profile for executing all services and UI on separate ports

### PR DESCRIPTION
**Pull request by kenfinnigan**
Add a `start` profile to start all 4 services at the same time, and then an npm server for the UI.

To make this work, some parts of web-application module were split to only be picked up when certain profiles are activated.

Each of the services also needed a CORS Filter added

By default, no `start` profile, will result in the same behavior of all services being deployed as WARs into TomEE

### CODE REVIEW

**Comment by karianna** [comment]
Does the UI need a Node JS server to run?  Wouldn't that run in a browser or is this for testing?

File: web-application/src/main/static/app/shared/endpoints.service.ts [hunk 1]
```diff
@@ -5,37 +5,40 @@ import {Endpoints} from "./endpoints";
...
 
-/**
- * This service is the doorway to endpoint discovery.
- * It contacts the 'only' known relative endpoint and retrieves a list of 'unknown' mapped endpoints.
- */
 @Injectable()
 export class EndpointsService {
 
-    map: {[name: string]: Endpoint} = {};
+    map: {[name: string]: Endpoint} =
```
Comment (AndyGee) [inline web-application/src/main/static/app/shared/endpoints.service.ts]: This doesn't feel good. Have we replaced the endpoint resolution app with a hard coded map, just to run in a node server?

Comment (AndyGee) [inline web-application/src/main/static/app/shared/endpoints.service.ts]: Or is this 'just' when running in node?

**Comment by kenfinnigan** [comment]
@karianna, I just used a Node JS server as there was already one setup in the package.json for the UI, plus it opens a browser to the correct URL for you which is nice.

@AndyGee, The endpoint resolution app is still present under normal circumstances. It's only when running with `-Pstart` that we use hard coded URLs in the UI

**Comment by OndrejM** [comment]
@karianna I'm not the author, but I believe that the reason for using Node.js is to demonstrate the idea of interoperability (see [minutes from the BOF](https://groups.google.com/d/msg/microprofile/P8NVZNPSxew/mGxilBevFQAJ) last week)
We want to experiment and open door to technologies outside Java EE and even outside JVM to integrate with MicroProfile-based microservices in a common way.

**Comment by karianna** [comment]
Fair enough, I was just caught out as the web apps I've been building recently just deliver a HTML5 app to the browser which executes/runs there.

**Pull request merged**